### PR TITLE
coprocessor/xeval: add ceil int function impl

### PR DESCRIPTION
**Pull request by ariesdevil**
This PR add the implementation of math function ceil_int to expression evaluator.

PTAL @XuHuaiyu @AndreMouche @lishihai9017 @andelf

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL @AndreMouche @andelf

File: src/coprocessor/xeval/builtin_math.rs [hunk 1]
```diff
@@ -44,9 +45,16 @@ impl Evaluator {
...
-        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    pub fn ceil_int(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        match d {
+            Datum::F64(i) => {
+                let result = i.ceil() as i64;
+                Ok(Datum::I64(result))
+            }
+            _ => invalid_type_error(&d, TYPE_FLOAT),
```
Comment (AndreMouche) [inline src/coprocessor/xeval/builtin_math.rs]: It seems we should return `Datum::None` when d is a None according mysql
```
mysql> SELECT CEILING(Null);
+---------------+
| CEILING(Null) |
+---------------+
|          NULL |
+---------------+
1 row in set (0.00 sec)

```

File: src/coprocessor/xeval/builtin_math.rs [hunk 2]
```diff
@@ -44,9 +45,17 @@ impl Evaluator {
         Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
     }
 
-    pub fn ceil_int(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
-        // TODO add impl
-        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    pub fn ceil_int(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
```
Comment (BusyJay) [inline src/coprocessor/xeval/builtin_math.rs]: To support `ceil_int`, we may need to support `eval_real` first.

Comment (BusyJay) [inline src/coprocessor/xeval/builtin_math.rs]: I made a mistake, looks like TiDB has handled it properly.

**Comment by ariesdevil** [comment]
@AkihiroSuda @BusyJay Could you help me rerun ci, gcc test passed but clang test stuck here https://travis-ci.org/pingcap/tikv/jobs/256762178#L2658

**Comment by AndreMouche** [comment]
Thank you for your contribution! @ariesdevil

**Pull request merged**